sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -102,16 +104,19 @@ def _csr_mean_variance_axis0(np.ndarray[floating, ndim=1, mode="c"] X_data,
 
     means = np.zeros(n_features, dtype=dtype)
     variances = np.zeros_like(means, dtype=dtype)
+    n_samples_arr = np.zeros_like(variances, dtype=dtype) + n_samples
 
     # counts[j] contains the number of samples where feature j is non-zero
     cdef np.ndarray[int, ndim=1] counts = np.zeros(n_features,
                                                    dtype=np.int32)
 
     for i in xrange(non_zero):
         col_ind = X_indices[i]
-        means[col_ind] += X_data[i]
-
-    means /= n_samples
+        if X_data[i] != X_data[i] and accepting_nan == True:
```

Comment:
`isnan` should be a C function, that should be more readable and as fast, isn't it?

Reply:
```python
x_i = X_data[i]
if isnan(x_i) and ignore_nan:
  n_samples_arr[col_ind] -= 1
else:
  means[col_ind] += x_i
```